### PR TITLE
Update links referencing git branches from master -> main

### LINKED ISSUE #8438: Broken links in MSBuild doc/sources
A number of links reference the old 'master' branch name.

I am opening a PR to update these to 'main', along with updating a couple of other broken links.

### PR DESCRIPTION
**Pull request by adr26**
Fix other broken links.

Fixes #8438

### Context

The primary branch for msbuild (and a number of other projects) has been migrating from 'master' to 'main'. A number of links reference the old 'master' branch name and are broken. 

### Changes Made

This PR updates those that have changed to 'main', along with updating a couple of other broken links.

### Testing

Relying on the CI tests :)

### Notes

It's my first MSBuild PR, apologies if there I've missed anything!

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/msbuild?pullRequest=8437) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/msbuild?pullRequest=8437) before we can accept your contribution.<br/><br/>:x: adr26 [sign now](https://cla.dotnetfoundation.org/dotnet/msbuild?pullRequest=8437)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/msbuild?pullRequest=8437) it.</sub>

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/msbuild?pullRequest=8437) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/msbuild?pullRequest=8437) before we can accept your contribution.<br/><br/>:x: adr26 [sign now](https://cla.dotnetfoundation.org/dotnet/msbuild?pullRequest=8437)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/msbuild?pullRequest=8437) it.</sub>

File: eng/common/native/CommonLibrary.psm1 [hunk 1]
```diff
@@ -230,7 +230,7 @@ function New-ScriptShim {
     # WinShimmer is a small .NET Framework program that creates .exe shims to bootstrapped programs
     # Many of the checks for installed programs expect a .exe extension for Windows tools, rather
     # than a .bat or .cmd file.
-    # Source: https://github.com/dotnet/arcade/tree/master/src/WinShimmer
+    # Source: https://github.com/dotnet/arcade/tree/main/src/WinShimmer
     if (-Not (Test-Path "$ShimDirectory\WinShimmer\winshimmer.exe")) {
       $InstallStatus = DownloadAndExtract -Uri "$BaseUri/windows/winshimmer/WinShimmer.zip" `
                                           -InstallDirectory $ShimDirectory\WinShimmer `
```
Comment (vlada-shubina) [inline eng/common/native/CommonLibrary.psm1:236]: changes to `eng/common` folder will be overwritten. If you'd like to contribute and make this change, please change it here: https://github.com/dotnet/arcade/blob/main/eng/common/native/CommonLibrary.psm1

File: .github/ISSUE_TEMPLATE/03_mybuildisbroken.md [hunk 2]
```diff
@@ -10,9 +10,9 @@ labels: needs-triage
 
 <!--
 * Here are some useful links to help you figure out what's wrong.
-* Our wiki: https://github.com/microsoft/msbuild/blob/master/documentation/wiki/Home.md
-* General Help: https://github.com/microsoft/msbuild/blob/master/documentation/wiki/Something's-wrong-in-my-build.md
-* Tips & Tricks: https://github.com/microsoft/msbuild/blob/master/documentation/wiki/MSBuild-Tips-%26-Tricks.md
+* Our documentation: https://github.com/dotnet/msbuild/blob/main/documentation/README.md
+* General Help: https://github.com/microsoft/msbuild/blob/main/documentation/wiki/Something's-wrong-in-my-build.md
+* Tips & Tricks: https://github.com/microsoft/msbuild/blob/main/documentation/wiki/MSBuild-Tips-%26-Tricks.md
```
Comment (vlada-shubina) [inline .github/ISSUE_TEMPLATE/03_mybuildisbroken.md:15]:
```suggestion
* General Help: https://github.com/dotnet/msbuild/blob/main/documentation/wiki/Something's-wrong-in-my-build.md
* Tips & Tricks: https://github.com/dotnet/msbuild/blob/main/documentation/wiki/MSBuild-Tips-%26-Tricks.md
```

**Review by vlada-shubina — COMMENTED** [review]
please do not apply changes to /eng/common folder

File: eng/common/native/CommonLibrary.psm1 (hunk 1 above)
Comment (adr26) [reply eng/common/native/CommonLibrary.psm1]: Thank you @vlada-shubina &mdash; I reverted these changes in [commit `d62c154447`](https://github.com/dotnet/msbuild/pull/8437/commits/d62c154447cf10d8c2b09d9b7e3f02c5c9643ee2), and have pushed that along with the change you suggested below (in [commit `00bdb0046c`](https://github.com/dotnet/msbuild/pull/8437/commits/00bdb0046c95dfacd8722ab3f306260feeefc20e)).

File: .github/ISSUE_TEMPLATE/03_mybuildisbroken.md (hunk 2 above)
Comment (adr26) [reply .github/ISSUE_TEMPLATE/03_mybuildisbroken.md]: Thank you @vlada-shubina &mdash; I have pushed this in [commit `00bdb0046c`](https://github.com/dotnet/msbuild/pull/8437/commits/00bdb0046c95dfacd8722ab3f306260feeefc20e), along with reverting the changes to `eng/common`.

**Review by rainersigwald — APPROVED** [review]
Thanks! Looks great with a couple of small changes.

**Review by vlada-shubina — APPROVED** [review]
Thanks!

**Review by Forgind — APPROVED** [review]

File: src/Shared/FileMatcher.cs [hunk 3]
```diff
@@ -2311,7 +2311,7 @@ private static bool IsFullFileSystemScan(int directoryPartIndex, int directoryPa
         /// Returns true if the given character is a valid drive letter.
         /// </summary>
         /// <remarks>
-        /// Copied from https://github.com/dotnet/corefx/blob/master/src/Common/src/System/IO/PathInternal.Windows.cs#L77-L83
+        /// Copied from https://github.com/dotnet/corefx/blob/master/src/Common/src/System/IO/PathInternal.Windows.cs#L53-L59
```
[the remark this answers is not in the document]
Comment (adr26) [reply src/Shared/FileMatcher.cs]: I saw [`dotnet/corefx`](https://github.com/dotnet/corefx) was archived / read-only now, so thought that this wouldn't move and hence just this fix was sufficient. However, I agree making it commit-specific is an extra layer of security and worthwhile!

I've pushed this in [commit `b6a65cb527`](https://github.com/dotnet/msbuild/pull/8437/commits/b6a65cb5279896602b449b0f64c988bf1995de00).

File: eng/common/internal-feed-operations.ps1 [hunk 4]
```diff
@@ -83,7 +83,7 @@ function SetupCredProvider {
   }
 }
 
-#Workaround for https://github.com/microsoft/msbuild/issues/4430
+#Workaround for https://github.com/dotnet/msbuild/issues/4430
```
[the remark this answers is not in the document]
Comment (adr26) [reply eng/common/internal-feed-operations.ps1]: I am really sorry, I managed to miss that one when cleaning out these changes: I've reverted this now in [commit `830d1ba33c`](https://github.com/dotnet/msbuild/pull/8437/commits/830d1ba33c983271030f139e6107cb4670122e0d).

**Pull request merged**